app.rb

```diff
@@ -101,4 +102,58 @@ def create_person
       nil
     end
   end
+
+  def save_people_data
+    File.write('people.json', JSON.pretty_generate(
+                                @people.map do |person|
+                                  if person.instance_of?(Teacher)
+                                    { name: person.name, age: person.age, parent_permission: person.parent_permission,
+                                      specialization: person.specialization }
+                                  else
+                                    { name: person.name, age: person.age, parent_permission: person.parent_permission }
+                                  end
+                                end
+                              ))
+  end
+
+  def save_books_data
+    File.write('books.json', JSON.pretty_generate(@books.map { |book| { title: book.title, author: book.author } }))
+  end
+
+  def save_rentals_data
+    File.write('rentals.json', JSON.pretty_generate(@rentals.map do |rental|
+                                                      { date: rental.date, book: rental.book.title,
+                                                        person: rental.person.name }
+                                                    end))
+  end
+
+  def load_books_data
+    return if File.zero?('books.json')
+
+    @books = JSON.parse(File.read('books.json')).map do |book|
+      Book.new(book['title'], book['author'])
+    end
+  end
+
+  def load_people_data
+    return if File.zero?('people.json')
+
+    @people = JSON.parse(File.read('people.json')).map do |person|
+      if person['specialization']
+        Teacher.new(person['age'], person['specialization'], person['name'])
+      else
+        Student.new(person['age'], person['name'], parent_permission: person['parent_permission'])
+      end
+    end
+  end
+
+  def load_rentals_data
+    return if File.zero?('rentals.json')
+
+    @rentals = JSON.parse(File.read('rentals.json')).map do |rental|
+      Rental.new(rental['date'], @books.find { |book| book.title == rental['book'] }, @people.find do |person|
+                                                                                        person.name == rental['person']
```

Comment:
- [ ] When I tried to run the app without the `JSON` files where data stored it crashed and it should work without JSON file and save the data and required in the requirements so please try to fix that 
<img width="1012" alt="Screenshot 2023-02-07 at 14 37 34" src="https://user-images.githubusercontent.com/61790587/217260346-957f97e5-1f8a-4023-833e-2636cb612d7e.png">

Reply:
@youmari it's strange because I'm not facing any issues on my part if you have time let's check with each other on a zoom call

Reply:
@Cluab We've talked on zoom